### DISCUSSION TITLE
Looking for a php filter or function for automating the "Regenerate CSS & Data"

### DISCUSSION THREAD

**Discussion by sinanisler** [elementor/elementor#19166] [Developers Community]
Help
Looking for a filter or function for automating the "Regenerate CSS & Data" button

I want to trigger this Regenerate CSS every time I save a elementor editor.

Does elementor have this type of php function, filter, action i can use ?
![image](https://user-images.githubusercontent.com/1686324/179171576-2ec8cef1-55b1-4956-9624-4e1e3c447242.png)

**Comment by Tygovanommen** [comment]
@sinanisler the following code allowes you to clear the cache:


```
// Check if Elementor installed and activated
if ( did_action( 'elementor/loaded' ) ) {
    // clear Elementor cache
    \Elementor\Plugin::instance()->files_manager->clear_cache();
}
```

**Comment by sinanisler** [comment]
hmmm i can trigger this with another filter or action.
nice.

didn't test it yet but 
thanks man.

**Comment by sinanisler** [comment]
after searching new hooks found this solution too

https://developers.elementor.com/docs/hooks/save-editor-data/

**Reply by JeroenKeijzer** [reply]
I have a similar problem and i was wondering if you use Litespeed Cache? I might have found the problem.

**Comment by RonR-WebDesign** [comment]
A plugin: "Clear Cache for Me" will automatically clear cache and, with the latest release, will also automatically run Regenerate CSS.  This will happen with plugin updates, page and post changes, and much more.  Check it out:  [HERE]( https://wordpress.org/plugins/clear-cache-for-widgets)
